addon/decompress.h

```diff
@@ -0,0 +1,31 @@
+#include <vector>
+
+#include "compression_worker.h"  // CompressionResult
+#include "zstd.h"
+
+CompressionResult decompress(const std::vector<uint8_t>& compressed) {
+    std::vector<uint8_t> decompressed;
+
+    using DCTX_Deleter = void (*)(ZSTD_DCtx*);
+
+    std::unique_ptr<ZSTD_DCtx, DCTX_Deleter> decompression_context(ZSTD_createDCtx(),
+                                                                   (DCTX_Deleter)ZSTD_freeDCtx);
+
+    ZSTD_inBuffer input = {compressed.data(), compressed.size(), 0};
+
+    while (input.pos < input.size) {
```

Comment:
Should we condition this on `input.pos < input.size || output.pos == output.size`?

Looking at the zstd docs, it seems that not doing so may lead to a cas where we haven't flushed all the data to the output stream, potentially leading to data loss

Reply:
Ah, nice catch!  Fixed.